tower-batch/src/semaphore.rs

```diff
@@ -34,10 +34,13 @@ impl Semaphore {
         }
     }
 
-    pub(crate) fn poll_acquire(&mut self, cx: &mut Context<'_>) -> Poll<()> {
+    pub(crate) fn poll_acquire(
+        &mut self,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<(), Box<dyn std::error::Error>>> {
         loop {
             self.state = match self.state {
-                State::Ready(_) => return Poll::Ready(()),
+                State::Ready(_) => return Poll::Ready(Ok(())),
```

Comment:
I was wrong, we don't need a `Result` here, because tokio 0.3.7 panics on error:
https://docs.rs/tokio/0.3.7/src/tokio/sync/semaphore.rs.html#143-149